src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1724,4 +1733,250 @@ void scaleInletFlows(EnergyPlusData &state, int const zoneNodeNum, int const spa
+{
+    auto &thisZoneEnergyDemand = state.dataZoneEnergyDemand->ZoneSysEnergyDemand(zoneNum);
+    auto &thisZoneMoistureDemand = state.dataZoneEnergyDemand->ZoneSysMoistureDemand(zoneNum);
+    Real64 sensibleRatio = 1.0;
+    Real64 latentRatio = 1.0;
+    switch (this->tstatControl) {
+    case DataZoneEquipment::ZoneEquipTstatControl::Ideal:
+        return;
+        break; // Do nothing
+    case DataZoneEquipment::ZoneEquipTstatControl::SingleSpace: {
+        Real64 controlSpaceFrac = this->spaces[this->controlSpaceNumber].fraction;
+        if (controlSpaceFrac > 0.0) {
+            if (thisZoneEnergyDemand.RemainingOutputRequired != 0.0) {
+                sensibleRatio = (state.dataZoneEnergyDemand->spaceSysEnergyDemand(this->controlSpaceIndex).RemainingOutputRequired /
+                                 thisZoneEnergyDemand.RemainingOutputRequired) /
+                                controlSpaceFrac;
+            }
+            if (thisZoneMoistureDemand.RemainingOutputRequired != 0.0) {
+                latentRatio = (state.dataZoneEnergyDemand->spaceSysMoistureDemand(this->controlSpaceIndex).RemainingOutputRequired /
+                               thisZoneMoistureDemand.RemainingOutputRequired) /
+                              controlSpaceFrac;
+            }
+        }
+    } break;
+    case DataZoneEquipment::ZoneEquipTstatControl::Maximum: {
+        int maxSpaceIndex = 0;
+        Real64 maxDeltaTemp = 0.0; // Only positive deltaTemps are relevant
+        Real64 maxSpaceFrac = 1.0;
+        for (auto &splitterSpace : this->spaces) {
+            Real64 spaceTemp =
+                state.dataZoneTempPredictorCorrector->spaceHeatBalance(splitterSpace.spaceIndex).T1; // Based on calcPredictedSystemLoad usage
+            Real64 spaceDeltaTemp = max((state.dataHeatBalFanSys->ZoneThermostatSetPointLo(zoneNum) - spaceTemp),
+                                        (spaceTemp - state.dataHeatBalFanSys->ZoneThermostatSetPointHi(zoneNum)));
+            if (spaceDeltaTemp > maxDeltaTemp) {
+                maxSpaceIndex = splitterSpace.spaceIndex;
+                maxSpaceFrac = splitterSpace.fraction;
+                maxDeltaTemp = spaceDeltaTemp;
+            }
+        }
+        if ((maxSpaceIndex > 0) && (maxSpaceFrac > 0.0)) {
+            if (thisZoneEnergyDemand.RemainingOutputRequired != 0.0) {
+                sensibleRatio = (state.dataZoneEnergyDemand->spaceSysEnergyDemand(maxSpaceIndex).RemainingOutputRequired /
+                                 thisZoneEnergyDemand.RemainingOutputRequired) /
+                                maxSpaceFrac;
+            }
+            if (thisZoneMoistureDemand.RemainingOutputRequired != 0.0) {
+                latentRatio = (state.dataZoneEnergyDemand->spaceSysMoistureDemand(maxSpaceIndex).RemainingOutputRequired /
+                               thisZoneMoistureDemand.RemainingOutputRequired) /
+                              maxSpaceFrac;
+            }
```

Comment:
It took me a minute to see why you were saving the original zone loads and I see now the space loads can be different depending on how the fraction is derived.

Reply:
Yeah, this was an attempt to avoid having *e v-e r-y* piece of HVAC equipment look to see if it should be controlling to the zone load or a specific space load. It' working, kindof, but the controls are not as precise as I expected. Room for improvement in the next cycle.